roles/v2/pool/src/lib/template_receiver/mod.rs

```diff
@@ -130,26 +131,30 @@ impl TemplateRx {
         }
     }
 
-    pub async fn send(self_: Arc<Mutex<Self>>, sv2_frame: StdFrame) -> Result<(), ()> {
+    pub async fn send(self_: Arc<Mutex<Self>>, sv2_frame: StdFrame) -> PoolResult<()> {
         let either_frame = sv2_frame.into();
-        let sender = self_.safe_lock(|self_| self_.sender.clone()).unwrap();
-        match sender.send(either_frame).await {
-            Ok(_) => Ok(()),
-            Err(_) => {
-                todo!()
-            }
-        }
+        let sender = self_
+            .safe_lock(|self_| self_.sender.clone())
+            .map_err(|e| PoolError::PoisonLock(e.to_string()))?;
+        sender.send(either_frame).await?;
+        Ok(())
     }
 
     async fn on_new_solution(self_: Arc<Mutex<Self>>, rx: Receiver<SubmitSolution<'static>>) {
+        let status_tx = self_.safe_lock(|s| s.status_tx.clone()).unwrap();
         while let Ok(solution) = rx.recv().await {
-            let sv2_frame: StdFrame =
+            let sv2_frame_res: Result<StdFrame, _> =
                 PoolMessages::TemplateDistribution(TemplateDistribution::SubmitSolution(solution))
-                    .try_into()
-                    .expect("Failed to convert solution to sv2 frame!");
-            Self::send(self_.clone(), sv2_frame)
-                .await
-                .expect("Failed to send solution!");
+                    .try_into();
+            match sv2_frame_res {
+                Ok(frame) => {
+                    handle_result!(status_tx, Self::send(self_.clone(), frame).await);
+                }
+                Err(_e) => {
+                    // return submit error
+                    todo!()
```

Comment:
Was this intentionally left?

Reply:
Yes, this should return a SubmitShares.Error message defined by the sv2 spec. it is in our backlog to implement this